### PR TITLE
Convert all queue endpoints + more tests + list response

### PR DESCRIPTION
**Pull request by brandur**
Continue API endpoint conversion work by converting all queue endpoints
to the new system and add requisite tests for all of them.

Another thing I added here is a generic `listResponse` so that list
endpoints start returning top level objects instead of raw arrays. This
is more compatible with the API endpoint framework for one, but from an
API design standpoint it's better because it's more forward compatible.
In case you ever want to add a new field to a list response, you don't
have any flexibility on a raw array because there's nowhere to put
anything new, but with a top-level object, you can add new keys to it.

### CODE REVIEW

**Comment by brandur** [comment]
@bgentry I was going to try and get all of the rest of them in one go, but I ended up having to add so much new supporting stuff that even just converting queues turned into a massive change, so I'm going to do the rest in two parts: queues, then everything else. Mind taking a look?

File: go.sum [hunk 1]
```diff
@@ -37,14 +37,23 @@ github.com/pmezard/go-difflib v1.0.0 h1:4DBwDE0NGyQoBHbLQYPwSUPoCMWR5BEzIk/f1lZb
 github.com/pmezard/go-difflib v1.0.0/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZNVY4sRDYZ/4=
 github.com/riverqueue/river v0.7.0 h1:STWnPn0APPKQkAjsJG5Q8NzFOxPXI/BKNAL6dPT7oI8=
 github.com/riverqueue/river v0.7.0/go.mod h1:ulIQc2U1mUTQNTnInytnVu4vhU9GJGeWReXtndvLBXI=
+github.com/riverqueue/river v0.8.0 h1:IBUIP9eZX/dkLQ3T+XNNk0Zi7iyUksZd4aHxQIFChOQ=
+github.com/riverqueue/river v0.8.0/go.mod h1:EHRbhqVXDpXQizFh4lndwswu53N0txITrLM2y3vOIF4=
```
Comment (bgentry) [inline go.sum:41]: Is this missing a `go mod tidy` or are we using both versions somehow?

**Review by bgentry — APPROVED** [review]
:shipit:

File: go.sum (hunk 1 above)
Comment (brandur) [reply go.sum]: Ah I just ran `go mod tidy` and it cleaned a few of these out.

**Comment by brandur** [comment]
Thanks!

**Pull request merged**